### PR TITLE
AV-196226 Support for Services with multiple ports

### PR DESCRIPTION
**Pull request by aaha97**

### CODE REVIEW

**Comment by aaha97** [comment]
build ako

**Comment by aaha97** [comment]
build ako

**Comment by aaha97** [comment]
build ako

**Review by pkoshtavmware — APPROVED** [review]
LGTM

File: ako-gateway-api/lib/lib.go [hunk 1]
```diff
@@ -79,3 +80,42 @@ func FindListenerStatusByName(name string, status []gatewayv1.ListenerStatus) in
...
+		utils.AviLog.Warnf("key: %s, msg: error while fetching service object: %s", key, err)
+		return intstr.IntOrString{}
+	}
+	if svcObj.Spec.Type == "NodePort" {
+		// Service of type NodePorts are not supported with tagertPort info. In such a case, the ports in the ingress must be strings
+		return intstr.IntOrString{}
+	}
+	for _, port := range svcObj.Spec.Ports {
+		// Iterate the ports and find the match for targetPort
+		if svcPort == port.Port {
```
Comment (akshayhavile) [inline ako-gateway-api/lib/lib.go:115]: @aaha97 While defining service, user can specify `portName` or `portNumber` so we should check both of it while returning targetPort for that service.

**Review by akshayhavile — APPROVED** [review]

**Pull request merged**